api/metrics-api-jaxrs-1.1/src/main/java/org/hawkular/metrics/api/jaxrs/handler/GaugeHandler.java

```diff
@@ -247,9 +253,34 @@ public Response findGaugeData(
                     percentiles = new Percentiles(Collections.<Double>emptyList());
                 }
 
-                return metricsService
-                        .findGaugeStats(metricId, timeRange.getStart(), timeRange.getEnd(), buckets,
-                                percentiles.getPercentiles())
+                Observable<TimeRange> observableTimeRange = Observable.just(timeRange);
+
+                if (Boolean.TRUE.equals(fromEarliest)) {
```

Comment:
What is this ```if```` block doing? You are not doing anything with the results. Am I missing something? Same question applies to corresponding method in GaugeHandler.